### ISSUE TITLE
CheckpointIO fails intermittently on moderate numbers of processors

### ISSUE THREAD

**Issue by roystgnr** [libMesh/libmesh#1169]
Running
```$ for num in $(seq 1 24); do LIBMESH_RUN="mpirun -np $num" make -j 24 check -C tests/ || break; done```

I get failures within -np 12 or so... at -np 10 on one run, at -np 8 on another, and so forth.  Changing the reading mesh from Mesh to ReplicatedMesh doesn't seem to help.

@jwpeterson, @friedmud, can you replicate this?

**Comment by roystgnr** [comment]
Specifically, I see:

```Attempted to utilize a checkpoint file with an incompatible mesh distribution!
[3] ../src/mesh/checkpoint_io.C, line 551, compiled Nov 22 2016 at 16:13:44```

**Comment by jwpeterson** [comment]
> On Nov 23, 2016, at 4:10 PM, roystgnr <notifications@github.com> wrote:
> 
> Specifically, I see:
> 
> [3] ../src/mesh/checkpoint_io.C, line 551, compiled Nov 22 2016 at 16:13:44```

How big is the mesh we're splitting up in that unit test?  My guess would be that Metis does something unexpected at one of those processor counts... But I won't be able to check it out until next week some time.

**Comment by jwpeterson** [comment]
I believe this specific issue was fixed by c43122ea2d31941b6749d935ba644466ffa8a945.  There may be others...
